### ISSUE TITLE
[Discussion] Guides/Documentation Improvements

### ISSUE THREAD

**Issue by Hamsterland** [Nihlus/Remora.Discord#216]
### Description

- [] Decide on documentation host (Continue with Wyam, Pages, Statiq etc...)
- [] Decide on what guides and tutorials to make

### Why This is Needed

To make Remora.Discord more approachable to new users who don't neccessarily have the time to slowly learn how it works like most of us have.

### Alternatives Considered

N/A

**Comment by MazeXP** [comment]
*Possible documentation tools*
- DocFX (requires GitHub Action to run on Windows)
- Statiq Docs (had some issues with API docs generation)

**Comment by AraHaan** [comment]
that is ok to use docfx with github actions on Windows, for pull requests it could also act as an documentation verifier as well which could verify that the docs do not have common errors / issues.

**Comment by Nihlus** [comment]
This is something that's sorely needed, and I really appreciate it! 

As for the tools of choice, I'm fine with whatever as long as it's open-source and cross-platform. No Windows-only stuff, please.

**Comment by Hamsterland** [comment]
I was a huge advocate for GitBook, but that did not support auto-generation of documention. However, I found this, which is very modern and looks promising!

https://github.com/webmaster442/BookGen

I'll play around with it a little and see how it works out.

**Comment by Hamsterland** [comment]
@MazeXP @Jax I think DocFX is the sensible choice here. Other options we've explored are either dead/deprecated (Sandcastle, Wyam) or too premature (Bookgen, Statiq). 

What do you think of DocFX?

**Comment by VelvetThePanda** [comment]
Heck. I'd love to write some articles then.

**Comment by Hamsterland** [comment]
That would be pretty great. You could really just start writing anywhere and we can import it later as DocFX uses markdown.

**Comment by VelvetThePanda** [comment]
Aye. I've written articles for D#+.

**Comment by Hamsterland** [comment]
Sweet. Let's just get @Nihlus final approval before jumping the gun.

**Comment by MazeXP** [comment]
DocFX 2.0 runs on NetFX.
One can run it with Mono on Linux basically

**Comment by Nihlus** [comment]
I was looking at DocFx in the past and really wanted to use it, but the framework-only builds were a blocker. I was mainly waiting for the 3.0 release which would address this, but perhaps we can start using it anyway. There are released preview versions - worth checking out!

**Comment by Nihlus** [comment]
Also, we should totally use the Discord theme.

**Comment by MazeXP** [comment]
DocFX 3,0 does not have the API reference generation inbuilt.

**Comment by Nihlus** [comment]
So assuming we use DocFx (which I really would like), the next step really needs to be a project plan - what goals do we have for the documentation, what do we want to document, in what format, etc etc - would you be willing to take the lead on that,  @Hamsterland?

**Comment by MazeXP** [comment]
For setting up DocFX in GitHub Actions it would be best to have it run on a windows machine.
Alternatively we could also set it up for Linux by using Mono with DocFX.

But as GitHub Actions does also provide Windows machines I do not see a problem about just
using Windows for the GitHub Action.
As I think the docs should actually be built by CI tasks.

As an alternative to GitHub Action it would also be possible to use AppVeyor for the docs generation.
(Free for public projects, 1 concurrent job, no limited usage time per month)

I did already some experiments with DocFX and Remora.Discord:
 - GitHub Action: https://github.com/MazeXP/Remora.Discord/blob/docs-improvement/.github/workflows/docs.yml
 - DocFX files: https://github.com/MazeXP/Remora.Discord/tree/docs-improvement/docs
 - Result: https://mazexp.github.io/Remora.Discord/index.html

**Comment by MazeXP** [comment]
Well, is it useful to be able to build docs locally?
For preview locally, as the articles would be written in markdown and most editors support previewing markdown.

**Comment by Nihlus** [comment]
I think so; testing a failing action is way faster if you can do it locally, as is experimenting with new features.

**Comment by MazeXP** [comment]
Would Mono be an option for you locally as DocFX 2 is written in NetFX?
DocFX 3 is sadly not making any progress regarding API Reference generation.

The only alternative would be Statiq.Docs when following issue gets fixed: https://github.com/statiqdev/Statiq.Docs/issues/52

**Comment by Nihlus** [comment]
Yeah, I'm fine using Mono; that said, if DocFX 3 doesn't seem like an option in the future, perhaps we should look elsewhere.

**Comment by MazeXP** [comment]
By new feature you mean a new article in the docs context, i guess?

**Comment by Nihlus** [comment]
Either a new article or the use of a new or different feature in DocFX itself, such as different themes or page generations.

**Comment by MazeXP** [comment]
Just came across https://docusaurus.io/. It looks good and easy to setup.
Maybe we could use it for Remora. It does not support API reference generation at all for .NET.
But to my extent I think it is not required for the beginning of a documentation.

**Comment by Nihlus** [comment]
I think I tried Docusaurus in the past with pretty good results. The lack of .NET API reference generation is a shame, though - we could, of course, always roll our own from the generated XML files and spit out a format Docusaurus recognizes.

**Comment by MazeXP** [comment]
How do you think about a separate branch that is then added via `git worktree` for documentation building.
